index.android.js

```diff
@@ -1,98 +1,126 @@
-var React = require('react');
-var ReactNative = require('react-native');
+import React, {
+  PropTypes,
+  Component
+} from 'react';
 
-var {
-  NativeModules,
-  StyleSheet,
+import {
   View,
+  Text,
+  StyleSheet,
+  NativeModules,
   TouchableHighlight,
-  Text
-} = ReactNative;
+} from 'react-native';
 
-var FBLoginManager = NativeModules.FBLoginManager;
+import itypeof from 'itypeof';
 
-var itypeof = require('itypeof');
+const { FBLoginManager } = NativeModules;
 
-var FBLogin = React.createClass({
-  childContextTypes: {
-    isLoggedIn: React.PropTypes.bool,
-    login: React.PropTypes.func,
-    logout: React.PropTypes.func,
-    props: React.PropTypes.object
+const styles = StyleSheet.create({
+  login: {
+    flex: 1,
+    backgroundColor: '#3B5998',
+    padding: 10,
+    alignItems: 'center'
   },
-  getChildContext: function () {
+  whiteFont: {
+    color: 'white'
+  }
+});
+
+const statics = {
+  loginText: 'Login with Facebook',
+  logoutText: 'Logout from Facebook'
+};
+
+class FBLogin extends Component {
+  constructor (props) {
+    super(props);
+
+    this.login = this.login.bind(this);
```

Comment:
instead of binding the context this way i think you can use the es6 arrow syntax

for eg. 

```js
login = (permissions) => {
 +    FBLoginManager.loginWithPermissions(
 +      permissions || this.props.permissions,
 +      (err,data) => this._handleEvent(err,data)
 +    );
 +  }
```